base/libgit2/libgit2.jl

```diff
@@ -329,42 +318,23 @@ end
 
 """ git reset [<committish>] [--] <pathspecs>... """
 function reset!(repo::GitRepo, committish::AbstractString, pathspecs::AbstractString...)
-    obj = revparse(repo, !isempty(committish) ? committish : Consts.HEAD_FILE)
+    obj = GitObject(repo, isempty(committish) ? Consts.HEAD_FILE : committish)
     # do not remove entries in the index matching the provided pathspecs with empty target commit tree
-    obj === nothing && throw(GitError(Error.Object, Error.ERROR, "`$committish` not found"))
-    try
-        head = reset!(repo, Nullable(obj), pathspecs...)
-        return head
-    finally
-        close(obj)
-    end
-    return head_oid(repo)
+    reset!(repo, Nullable(obj), pathspecs...)
 end
 
-""" git reset [--soft | --mixed | --hard] <commit> """
-function reset!(repo::GitRepo, commit::GitHash, mode::Cint = Consts.RESET_MIXED)
-    obj = get(GitUnknownObject, repo, commit)
-    # object must exist for reset
-    obj === nothing && throw(GitError(Error.Object, Error.ERROR, "Commit `$(string(commit))` object not found"))
-    try
-        head = reset!(repo, obj, mode)
-        return head
-    finally
-        close(obj)
-    end
-    return head_oid(repo)
+""" git reset [--soft | --mixed | --hard] <id> """
+function reset!(repo::GitRepo, id::GitHash, mode::Cint = Consts.RESET_MIXED)
+    reset!(repo, GitObject(repo, id), mode)
 end
 
 """ git cat-file <commit> """
-function cat{T<:GitObject}(repo::GitRepo, ::Type{T}, object::AbstractString)
-    obj_id = revparseid(repo, object)
-    iszero(obj_id) && return nothing
-
-    obj = get(T, repo, obj_id)
+function cat(repo::GitRepo, spec)
+    obj = GitObject(repo, spec)
     if isa(obj, GitBlob)
-        return unsafe_string(convert(Ptr{UInt8}, content(obj)))
+        content(obj)
     else
-        return nothing
+        nothing
```

Comment:
If the resulting object is a commit, tag or tree. Arguably this is another function that could be removed altogether.